src/Channel.js

```diff
@@ -1,54 +1,145 @@
-import {useState} from 'react'
+import {useState, useEffect} from 'react'
 import {supabase} from './supabaseClient'
 
 export default function Account({session}) {
 	const [loading, setLoading] = useState(false)
+	const [deleting, setDeleting] = useState(false)
 	const [name, setName] = useState('')
 	const [slug, setSlug] = useState('')
+	const [channel, setChannel] = useState('')
 
-	async function saveChannel({name, slug}) {
+	useEffect(() => {
+		findChannel()
+	}, [session])
+
+	async function findChannel() {
 		try {
 			setLoading(true)
 			const user = supabase.auth.user()
+			let {data, error, status} = await supabase
+				.from('channels')
+				.select(`*`)
+				.eq('user_id', user.id)
+				.single()
+			if (error && status !== 406) throw error
+			console.log('found channel', data)
+			if (data) setChannel(data)
+			if (!data) setChannel(null)
+		} catch (error) {
+			console.error(error.message)
+		} finally {
+			setLoading(false)
+		}
+	}
 
-			const updates = {
-				// id: 1111,
-				name,
-				slug,
-				user_id: user.id,
-				updated_at: new Date(),
-			}
-
-			let {error} = await supabase.from('channels').insert(updates)
+	async function createChannel({name, slug}) {
+		try {
+			setLoading(true)
+			const user = supabase.auth.user()
+			let {data, error} = await supabase
+				.from('channels')
+				.insert({name, slug, user_id: user.id, updated_at: new Date()})
+			if (error) throw error
+			setChannel(data)
+		} catch (error) {
+			console.error(error.message)
+		} finally {
+			setLoading(false)
+		}
+	}
 
-			if (error) {
-				throw error
-			}
+	async function updateChannel({name, slug}) {
+		try {
+			setLoading(true)
+			let {error} = await supabase
+				.from('channels')
+				.upsert({id: channel.id, name, slug, updated_at: new Date()})
```

Comment:
maybe the date can/should be a `postgresql` thing?
https://www.postgresql.org/docs/9.1/functions-datetime.html
https://stackoverflow.com/questions/1035980/update-timestamp-when-row-is-updated-in-postgresql

Reply:
yea good idea, created another issue for it https://github.com/internet4000/radio4000-cms/issues/7